src/v1.ts

```diff
@@ -0,0 +1,282 @@
+import {
+  PublicClient,
+  Account,
+  Address,
+  createPublicClient,
+  WalletClient,
+  createWalletClient,
+  Chain,
+  Transport,
+  http,
+  encodeFunctionData,
+} from 'viem';
+import { mainnet } from 'viem/chains';
+import { ADDRESS, ABI } from './contracts/v1.js';
+import { getV1DelegationType } from './utils.js';
+
+export type V1Delegation = {
+  type: string;
+  vault: `0x${string}`;
+  delegate: `0x${string}`;
+  contract: `0x${string}`;
+  tokenId: number;
+};
+
+export const DelegateV1 = class {
+  publicClient: PublicClient;
+  walletClient: WalletClient;
+  account?: Account | Address;
+  contractConfig = {
+    address: ADDRESS,
+    abi: ABI,
+  };
+
+  constructor(chain: Chain = mainnet, userTransport: Transport = http(), account?: Account | Address) {
```

Comment:
Would be nice to make `chain` optional and change the order to:

```
constructor(
    userTransport: Transport = http(), 
    chain: Chain = mainnet, 
    account?: Account | Address
)
```

Reply:
I like that, good call.